support-lambdas/acquisitions-firehose-transformer/src/main/scala/com/gu/acquisitionFirehoseTransformer/GBPConversionService.scala

```diff
@@ -0,0 +1,50 @@
+package com.gu.acquisitionFirehoseTransformer
+
+
+import com.gu.i18n.Currency
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+import org.scanamo.generic.auto._
+import org.scanamo.syntax._
+import org.scanamo.{Scanamo, Table}
+import software.amazon.awssdk.auth.credentials.{AwsCredentialsProviderChain, EnvironmentVariableCredentialsProvider, ProfileCredentialsProvider}
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+
+trait GBPConversionService {
+  def convert(currency: Currency, amount: Double, dateTime: DateTime): Either[String,Double]
+}
+
+object GBPConversionServiceImpl extends GBPConversionService {
+  private case class ConversionData(base: String, eventDate: String, rates: Map[String,Float])
+
+  private val providerChain = AwsCredentialsProviderChain.builder()
+    .addCredentialsProvider(EnvironmentVariableCredentialsProvider.create())  // For running in lambda
+    .addCredentialsProvider(ProfileCredentialsProvider.create("membership"))  // For running locally
+    .build()
+
+  private val dynamoDb = DynamoDbClient.builder()
+    .credentialsProvider(providerChain)
+    .region(Region.EU_WEST_1)
+    .build()
+
+  private val scanamo: Scanamo = Scanamo(dynamoDb)
+  private val table: Table[ConversionData] = Table("fixer-io-cache")
+
+  def convert(currency: Currency, amount: Double, dateTime: DateTime): Either[String,Double] = {
+    val date = ISODateTimeFormat.date().print(dateTime)
+    scanamo
+      .exec(
```

Comment:
this query result could be cached (if it succeeds) but I don't think it matters much...

Reply:
Oh I missed this when I left my comment about caching! Yeah, I guess if caching is awkward then it's not worth it. Acquisitions is of the order of 1000s a day so it's not too much stress on the db